x-pack/plugins/uptime/public/components/synthetics/check_steps/step_duration.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as React from 'react';
+import { EuiButtonEmpty, EuiPopover } from '@elastic/eui';
+import { useMemo, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import numeral from '@elastic/numeral';
+import { JourneyStep } from '../../../../common/runtime_types';
+import { StepFieldTrend } from './waterfall_marker_trend';
+
+interface Props {
+  step: JourneyStep;
+}
+
+export const StepDuration = ({ step }: Props) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const component = useMemo(
+    () => (
+      <StepFieldTrend
+        step={step}
+        field={'synthetics.step.duration.us'}
+        title={STEP_DURATION_TREND}
+      />
+    ),
+    [step]
+  );
+
+  if (step.synthetics.step?.status === 'skipped') {
+    return <span>--</span>;
+  }
+
+  const button = (
+    <EuiButtonEmpty onMouseEnter={() => setIsOpen(true)} iconType="visArea">
+      {i18n.translate('xpack.uptime.synthetics.step.duration', {
+        defaultMessage: '{value} sec',
+        values: {
+          value: numeral(step.synthetics.step?.duration.us! / (1000 * 1000)).format('00.0'),
```

Comment:
Could we extract this `number` literal computation to a named constant that's computed outside the function?

```typescript
const MICRO_TO_SECOND = 1000 * 1000;
```

```suggestion
          value: numeral(step.synthetics.step?.duration.us! / (MICRO_TO_SECOND)).format('00.0'),
```

Reply:
Done in [`b3eab1b` (#116266)](https://github.com/elastic/kibana/pull/116266/commits/b3eab1be59729f4749290875a78094383a99d1af)